bin/server/cmd/root.go

```diff
@@ -91,6 +93,9 @@ var (
 	readonly bool
 	demo     bool
 	debug    bool
+	// pgURL must follow PostgreSQL connection URIs pattern.
+	// https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING
+	pgURL string
```

Comment:
FWIW, I did some research while reviewing this, that's why I was asking to provide the example as I didn't make up my mind before.

Many software does use separate flags. e.g.

GitLab does separate them
https://docs.gitlab.com/omnibus/settings/database.html#configure-postgresql-block

Sourcegraph also separates them
https://docs.sourcegraph.com/admin/external_services/postgres

So is temporal
https://docs.temporal.io/docs/server/configuration/#persistence
https://docs.temporal.io/docs/server/versions-and-dependencies/#upgrading-mysql--postgresql-schema

While I am also in the Go DSN / JDBC camp to put all into a single connection string like how this PR is implemented.

Reply:
My personal preference is to use separate flags instead of taking a whole DSN string for a better user experience. Users don't need to look up the format of Postgres DSN.

Reply:
If we configure all this options in a config file, said a yaml or toml file, I'll prefer to use separate flags, e.g.,

```toml
[metadata.external]
enable = true
type = "postgresql"
user = "..."
passwd = "..."
# ...
```

However, it's in the command line and the parts are grouped to be present or absent (whether to provide an external pgURL is optional). Thus, if we flatten all flags to the command line it'll be quite clumsy and the composition is brittle.

Reply:
The downside of having separate flags is the user needs to learn to feed the values to the appropriate flags respectively.

Our situation is worse, e.g. we already have port, frontend-port, then we will have pg-port.

Reply:
@d-bytebase is it possible for `cobra` to group flags? If it's possible, then I can try to make an alternative for preview.

Reply:
TBH, though it's a bit struggle, I prefer the current way to have a single flag.

Reply:
Let's take a try for now.